pkg/client/key/key.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 IBM Corp
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package key
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/errors"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM-Cloud/power-go-client/power/client/p_cloud_tenants_ssh_keys"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+)
+
+type Client struct {
+	session    *ibmpisession.IBMPISession
+	client     *instance.IBMPIKeyClient
+	instanceID string
+}
+
+func NewClient(sess *ibmpisession.IBMPISession, powerinstanceid string) *Client {
+	c := &Client{
+		session:    sess,
+		instanceID: powerinstanceid,
+	}
+	c.client = instance.NewIBMPIKeyClient(sess, powerinstanceid)
+	return c
+}
+
+func (c *Client) Get(id string) (*models.SSHKey, error) {
+	return c.client.Get(id, c.instanceID)
+}
+
+func (c *Client) Create(name, sshKey string) (*models.SSHKey, *models.SSHKey, error) {
+	return c.client.Create(name, sshKey, c.instanceID)
+}
+
+func (c *Client) Delete(id string) error {
+	return c.client.Delete(id, c.instanceID)
+}
+
+// Get Keys...
+//TODO:Replace Getall function with upstream code
+func (c *Client) GetAll() (*models.SSHKeys, error) {
+
+	var tenantid = c.session.UserAccount
+	params := p_cloud_tenants_ssh_keys.NewPcloudTenantsSshkeysGetallParamsWithTimeout(helpers.PIGetTimeOut).WithTenantID(tenantid)
+	resp, err := c.session.Power.PCloudTenantsSSHKeys.PcloudTenantsSshkeysGetall(params, ibmpisession.NewAuth(c.session, c.instanceID))
+
+	if err != nil || resp == nil || resp.Payload == nil {
+		return nil, fmt.Errorf(errors.GetPIKeyOperationFailed, c.instanceID, err)
+	}
+	return resp.Payload, nil
+}
+
+func (c *Client) GetAllPurgeable(before, since time.Duration, expr string) ([]string, error) {
+	keys, err := c.GetAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the list of instances: %v", err)
+	}
+
+	var keysMatched []string
+	for _, key := range keys.SSHKeys {
+		if expr != "" {
+			if r, _ := regexp.Compile(expr); !r.MatchString(*key.Name) {
+				continue
+			}
+		}
```

Comment:
as I mentioned above we shouldn't entertain the empty expr string

Reply:
ok..